packages/sourcecred/src/plugins/discord/createGraph.js

```diff
@@ -320,48 +322,116 @@ export function _createGraphFromMessages(
       channelId,
       channelParentId,
     } = graphMessage;
-    if (author) {
-      wg.graph.addNode(memberNode(author));
-      wg.graph.addEdge(authorsMessageEdge(message, author));
-    }
-    wg.graph.addNode(messageNode(message, guildId, channelName));
 
+    let messageWeight = 0;
     for (const {reaction, reactingMember} of reactions) {
-      const node = reactionNode(reaction, message.timestampMs, guildId);
-      wg.weights.nodeWeights.set(
-        node.address,
-        reactionWeight(
-          weights,
-          message,
-          reaction,
-          reactingMember,
-          propsChannels,
-          reactions,
-          channelParentId
-        )
-      );
-      wg.graph.addNode(node);
-      wg.graph.addNode(memberNode(reactingMember));
-      wg.graph.addEdge(reactsToEdge(reaction, message));
-      wg.graph.addEdge(
-        addsReactionEdge(reaction, reactingMember, message.timestampMs)
+      const weight = reactionWeight(
```

Comment:
lots of args : /  ... maybe reactionWeight should just take {weights, graphMessage}

I see you're just refactoring what already existed, so grain of salt.

Reply:
Passing graphMessage would take the params down from 7 to 5... Usually what we do is turn a long list of params into an object, so then they at least have names identifying them, instead of mere param order.

Reply:
+1 options object, non-blocking